Binary files libmtp-1.1.14/._lsusb-canon-canoscan.txt and libmtp-1.1.14-new/._lsusb-canon-canoscan.txt differ
Binary files libmtp-1.1.14/._mtp-detect-acer-e380.txt and libmtp-1.1.14-new/._mtp-detect-acer-e380.txt differ
Binary files libmtp-1.1.14/._mtp-detect-acer-iconia-tab-a100-id2.txt and libmtp-1.1.14-new/._mtp-detect-acer-iconia-tab-a100-id2.txt differ
Binary files libmtp-1.1.14/._mtp-detect-acer-iconia-tab-a700.txt and libmtp-1.1.14-new/._mtp-detect-acer-iconia-tab-a700.txt differ
Binary files libmtp-1.1.14/._mtp-detect-acer-liquid-zest-plus.txt and libmtp-1.1.14-new/._mtp-detect-acer-liquid-zest-plus.txt differ
Binary files libmtp-1.1.14/._mtp-detect-acer-z220.txt and libmtp-1.1.14-new/._mtp-detect-acer-z220.txt differ
Binary files libmtp-1.1.14/._mtp-detect-alcatel-one-touch.txt and libmtp-1.1.14-new/._mtp-detect-alcatel-one-touch.txt differ
Binary files libmtp-1.1.14/._mtp-detect-amazon-fire-hd8.txt and libmtp-1.1.14-new/._mtp-detect-amazon-fire-hd8.txt differ
Binary files libmtp-1.1.14/._mtp-detect-amazon-fire-phone.txt and libmtp-1.1.14-new/._mtp-detect-amazon-fire-phone.txt differ
Binary files libmtp-1.1.14/._mtp-detect-amazon-kindle-fire-5.txt and libmtp-1.1.14-new/._mtp-detect-amazon-kindle-fire-5.txt differ
Binary files libmtp-1.1.14/._mtp-detect-archos-5.txt and libmtp-1.1.14-new/._mtp-detect-archos-5.txt differ
Binary files libmtp-1.1.14/._mtp-detect-archos-5-v1.txt and libmtp-1.1.14-new/._mtp-detect-archos-5-v1.txt differ
Binary files libmtp-1.1.14/._mtp-detect-archos-80g9.txt and libmtp-1.1.14-new/._mtp-detect-archos-80g9.txt differ
Binary files libmtp-1.1.14/._mtp-detect-archos-a504.txt and libmtp-1.1.14-new/._mtp-detect-archos-a504.txt differ
Binary files libmtp-1.1.14/._mtp-detect-archos-neon.txt and libmtp-1.1.14-new/._mtp-detect-archos-neon.txt differ
Binary files libmtp-1.1.14/._mtp-detect-asus-me173x.txt and libmtp-1.1.14-new/._mtp-detect-asus-me173x.txt differ
Binary files libmtp-1.1.14/._mtp-detect-asus-TF201-google.txt and libmtp-1.1.14-new/._mtp-detect-asus-TF201-google.txt differ
Binary files libmtp-1.1.14/._mtp-detect-asus-tx201la.txt and libmtp-1.1.14-new/._mtp-detect-asus-tx201la.txt differ
Binary files libmtp-1.1.14/._mtp-detect-asus-zenfone2.txt and libmtp-1.1.14-new/._mtp-detect-asus-zenfone2.txt differ
Binary files libmtp-1.1.14/._mtp-detect-asus-zenfone-4.txt and libmtp-1.1.14-new/._mtp-detect-asus-zenfone-4.txt differ
Binary files libmtp-1.1.14/._mtp-detect-asus-zenfone-6.txt and libmtp-1.1.14-new/._mtp-detect-asus-zenfone-6.txt differ
Binary files libmtp-1.1.14/._mtp-detect-bq-aquaris-e45-ubuntu-editon.txt and libmtp-1.1.14-new/._mtp-detect-bq-aquaris-e45-ubuntu-editon.txt differ
Binary files libmtp-1.1.14/._mtp-detect-bq-aquaris-m55.txt and libmtp-1.1.14-new/._mtp-detect-bq-aquaris-m55.txt differ
Binary files libmtp-1.1.14/._mtp-detect-casio-gzone-commando-c771.txt and libmtp-1.1.14-new/._mtp-detect-casio-gzone-commando-c771.txt differ
Binary files libmtp-1.1.14/._mtp-detect-cowon-iaudio-s9.txt and libmtp-1.1.14-new/._mtp-detect-cowon-iaudio-s9.txt differ
Binary files libmtp-1.1.14/._mtp-detect-creative-zen-V-plus.txt and libmtp-1.1.14-new/._mtp-detect-creative-zen-V-plus.txt differ
Binary files libmtp-1.1.14/._mtp-detect-dexp-ixion-xl145.txt and libmtp-1.1.14-new/._mtp-detect-dexp-ixion-xl145.txt differ
Binary files libmtp-1.1.14/._mtp-detect-google-pixel-c-adb.txt and libmtp-1.1.14-new/._mtp-detect-google-pixel-c-adb.txt differ
Binary files libmtp-1.1.14/._mtp-detect-gopro-hero5-black.txt and libmtp-1.1.14-new/._mtp-detect-gopro-hero5-black.txt differ
Binary files libmtp-1.1.14/._mtp-detect-haier-ibiza-rhapsody.txt and libmtp-1.1.14-new/._mtp-detect-haier-ibiza-rhapsody.txt differ
Binary files libmtp-1.1.14/._mtp-detect-hp-touchpad.txt and libmtp-1.1.14-new/._mtp-detect-hp-touchpad.txt differ
Binary files libmtp-1.1.14/._mtp-detect-htc-626.log and libmtp-1.1.14-new/._mtp-detect-htc-626.log differ
Binary files libmtp-1.1.14/._mtp-detect-htc-oneplusone.txt and libmtp-1.1.14-new/._mtp-detect-htc-oneplusone.txt differ
Binary files libmtp-1.1.14/._mtp-detect-htc-one-v.txt and libmtp-1.1.14-new/._mtp-detect-htc-one-v.txt differ
Binary files libmtp-1.1.14/._mtp-detect-iriver-clix2.txt and libmtp-1.1.14-new/._mtp-detect-iriver-clix2.txt differ
Binary files libmtp-1.1.14/._mtp-detect-iriver-t20.txt and libmtp-1.1.14-new/._mtp-detect-iriver-t20.txt differ
Binary files libmtp-1.1.14/._mtp-detect-kobo-arc7-hd.log and libmtp-1.1.14-new/._mtp-detect-kobo-arc7-hd.log differ
Binary files libmtp-1.1.14/._mtp-detect-lenovo-ideapad-A2109.txt and libmtp-1.1.14-new/._mtp-detect-lenovo-ideapad-A2109.txt differ
Binary files libmtp-1.1.14/._mtp-detect-lenovo-k900.txt and libmtp-1.1.14-new/._mtp-detect-lenovo-k900.txt differ
Binary files libmtp-1.1.14/._mtp-detect-lenovo-lifetab.txt and libmtp-1.1.14-new/._mtp-detect-lenovo-lifetab.txt differ
Binary files libmtp-1.1.14/._mtp-detect-lenovo-s660.txt and libmtp-1.1.14-new/._mtp-detect-lenovo-s660.txt differ
Binary files libmtp-1.1.14/._mtp-detect-lenovo-vibe-k5-note.txt and libmtp-1.1.14-new/._mtp-detect-lenovo-vibe-k5-note.txt differ
Binary files libmtp-1.1.14/._mtp-detect-lenovo-vibe-z2.txt and libmtp-1.1.14-new/._mtp-detect-lenovo-vibe-z2.txt differ
Binary files libmtp-1.1.14/._mtp-detect-lg-v10.txt and libmtp-1.1.14-new/._mtp-detect-lg-v10.txt differ
Binary files libmtp-1.1.14/._mtp-detect-logik-dax-dab.txt and libmtp-1.1.14-new/._mtp-detect-logik-dax-dab.txt differ
Binary files libmtp-1.1.14/._mtp-detect-maxfield-g-flash-ng-1GB.txt and libmtp-1.1.14-new/._mtp-detect-maxfield-g-flash-ng-1GB.txt differ
Binary files libmtp-1.1.14/._mtp-detect-motorola-droid-turbo-2.txt and libmtp-1.1.14-new/._mtp-detect-motorola-droid-turbo-2.txt differ
Binary files libmtp-1.1.14/._mtp-detect-motorola-razr2-v8.txt and libmtp-1.1.14-new/._mtp-detect-motorola-razr2-v8.txt differ
Binary files libmtp-1.1.14/._mtp-detect-motorola-v3m-verizon.txt and libmtp-1.1.14-new/._mtp-detect-motorola-v3m-verizon.txt differ
Binary files libmtp-1.1.14/._mtp-detect-motorola-v750.txt and libmtp-1.1.14-new/._mtp-detect-motorola-v750.txt differ
Binary files libmtp-1.1.14/._mtp-detect-motorola-xoom.txt and libmtp-1.1.14-new/._mtp-detect-motorola-xoom.txt differ
Binary files libmtp-1.1.14/._mtp-detect-motorola-xt1524.txt and libmtp-1.1.14-new/._mtp-detect-motorola-xt1524.txt differ
Binary files libmtp-1.1.14/._mtp-detect-nokia-300.txt and libmtp-1.1.14-new/._mtp-detect-nokia-300.txt differ
Binary files libmtp-1.1.14/._mtp-detect-nokia-5800-2.txt and libmtp-1.1.14-new/._mtp-detect-nokia-5800-2.txt differ
Binary files libmtp-1.1.14/._mtp-detect-nokia-6220.txt and libmtp-1.1.14-new/._mtp-detect-nokia-6220.txt differ
Binary files libmtp-1.1.14/._mtp-detect-nokia-6500c.txt and libmtp-1.1.14-new/._mtp-detect-nokia-6500c.txt differ
Binary files libmtp-1.1.14/._mtp-detect-nokia-n81.txt and libmtp-1.1.14-new/._mtp-detect-nokia-n81.txt differ
Binary files libmtp-1.1.14/._mtp-detect-nokia-n85.txt and libmtp-1.1.14-new/._mtp-detect-nokia-n85.txt differ
Binary files libmtp-1.1.14/._mtp-detect-nokia-n95.txt and libmtp-1.1.14-new/._mtp-detect-nokia-n95.txt differ
Binary files libmtp-1.1.14/._mtp-detect-nokia-n97.txt and libmtp-1.1.14-new/._mtp-detect-nokia-n97.txt differ
Binary files libmtp-1.1.14/._mtp-detect-oneplus-one3003.txt and libmtp-1.1.14-new/._mtp-detect-oneplus-one3003.txt differ
Binary files libmtp-1.1.14/._mtp-detect-oneplus-one-two-a2005.txt and libmtp-1.1.14-new/._mtp-detect-oneplus-one-two-a2005.txt differ
Binary files libmtp-1.1.14/._mtp-detect-panasonic-p905i.txt and libmtp-1.1.14-new/._mtp-detect-panasonic-p905i.txt differ
Binary files libmtp-1.1.14/._mtp-detect-pegatron-chagall.txt and libmtp-1.1.14-new/._mtp-detect-pegatron-chagall.txt differ
Binary files libmtp-1.1.14/._mtp-detect-philips-psa610.txt and libmtp-1.1.14-new/._mtp-detect-philips-psa610.txt differ
Binary files libmtp-1.1.14/._mtp-detect-philips-sa6025.txt and libmtp-1.1.14-new/._mtp-detect-philips-sa6025.txt differ
Binary files libmtp-1.1.14/._mtp-detect-philips-w6610.txt and libmtp-1.1.14-new/._mtp-detect-philips-w6610.txt differ
Binary files libmtp-1.1.14/._mtp-detect-pioneer-xmp3.txt and libmtp-1.1.14-new/._mtp-detect-pioneer-xmp3.txt differ
Binary files libmtp-1.1.14/._mtp-detect-pocket-tunes.txt and libmtp-1.1.14-new/._mtp-detect-pocket-tunes.txt differ
Binary files libmtp-1.1.14/._mtp-detect-redmi-hm1s.txt and libmtp-1.1.14-new/._mtp-detect-redmi-hm1s.txt differ
Binary files libmtp-1.1.14/._mtp-detect-redmi-note.txt and libmtp-1.1.14-new/._mtp-detect-redmi-note.txt differ
Binary files libmtp-1.1.14/._mtp-detect-samsung-gn-jb.txt and libmtp-1.1.14-new/._mtp-detect-samsung-gn-jb.txt differ
Binary files libmtp-1.1.14/._mtp-detect-samsung-gt-b2700.txt and libmtp-1.1.14-new/._mtp-detect-samsung-gt-b2700.txt differ
Binary files libmtp-1.1.14/._mtp-detect-samsung-gt-i9070.txt and libmtp-1.1.14-new/._mtp-detect-samsung-gt-i9070.txt differ
Binary files libmtp-1.1.14/._mtp-detect-samsung-gt-i9300.txt and libmtp-1.1.14-new/._mtp-detect-samsung-gt-i9300.txt differ
Binary files libmtp-1.1.14/._mtp-detect-samsung-yp-p3.txt and libmtp-1.1.14-new/._mtp-detect-samsung-yp-p3.txt differ
Binary files libmtp-1.1.14/._mtp-detect-samsung-yp-q1.txt and libmtp-1.1.14-new/._mtp-detect-samsung-yp-q1.txt differ
Binary files libmtp-1.1.14/._mtp-detect-samsung-yp-u4.txt and libmtp-1.1.14-new/._mtp-detect-samsung-yp-u4.txt differ
Binary files libmtp-1.1.14/._mtp-detect-samsung-yp-z3.txt and libmtp-1.1.14-new/._mtp-detect-samsung-yp-z3.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sandisk-sansa-c240.txt and libmtp-1.1.14-new/._mtp-detect-sandisk-sansa-c240.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sandisk-sansa-c250.txt and libmtp-1.1.14-new/._mtp-detect-sandisk-sansa-c250.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sandisk-sansa-e250.txt and libmtp-1.1.14-new/._mtp-detect-sandisk-sansa-e250.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sandisk-sansa-fuze.txt and libmtp-1.1.14-new/._mtp-detect-sandisk-sansa-fuze.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sirius-stiletto2.txt and libmtp-1.1.14-new/._mtp-detect-sirius-stiletto2.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sonyericsson-c905.txt and libmtp-1.1.14-new/._mtp-detect-sonyericsson-c905.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sonyericsson-t700.txt and libmtp-1.1.14-new/._mtp-detect-sonyericsson-t700.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sonyericsson-w760i.txt and libmtp-1.1.14-new/._mtp-detect-sonyericsson-w760i.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sonyericsson-w890i.txt and libmtp-1.1.14-new/._mtp-detect-sonyericsson-w890i.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sonyericsson-xperia-mk16i.txt and libmtp-1.1.14-new/._mtp-detect-sonyericsson-xperia-mk16i.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sony-walkman-nwz-e354.txt and libmtp-1.1.14-new/._mtp-detect-sony-walkman-nwz-e354.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sony-walkman-nwz-e436f.txt and libmtp-1.1.14-new/._mtp-detect-sony-walkman-nwz-e436f.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sony-walkman-nwz-s716f.txt and libmtp-1.1.14-new/._mtp-detect-sony-walkman-nwz-s716f.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sony-walkman-nwz-s765.txt and libmtp-1.1.14-new/._mtp-detect-sony-walkman-nwz-s765.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sony-walkman-nwz-w202.txt and libmtp-1.1.14-new/._mtp-detect-sony-walkman-nwz-w202.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sony-xperia-m2.txt and libmtp-1.1.14-new/._mtp-detect-sony-xperia-m2.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sony-xperia-sov31.txt and libmtp-1.1.14-new/._mtp-detect-sony-xperia-sov31.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sony-xperia-xa.txt and libmtp-1.1.14-new/._mtp-detect-sony-xperia-xa.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sony-xperia-x.log and libmtp-1.1.14-new/._mtp-detect-sony-xperia-x.log differ
Binary files libmtp-1.1.14/._mtp-detect-sony-xperia-z4-aqua-dual.txt and libmtp-1.1.14-new/._mtp-detect-sony-xperia-z4-aqua-dual.txt differ
Binary files libmtp-1.1.14/._mtp-detect-sony-xperia-z5.txt and libmtp-1.1.14-new/._mtp-detect-sony-xperia-z5.txt differ
Binary files libmtp-1.1.14/._mtp-detect-thalia-tolino-tab7.txt and libmtp-1.1.14-new/._mtp-detect-thalia-tolino-tab7.txt differ
Binary files libmtp-1.1.14/._mtp-detect-toshiba-gigabeat-s.txt and libmtp-1.1.14-new/._mtp-detect-toshiba-gigabeat-s.txt differ
Binary files libmtp-1.1.14/._mtp-detect-trekstor-vibez.txt and libmtp-1.1.14-new/._mtp-detect-trekstor-vibez.txt differ
Binary files libmtp-1.1.14/._mtp-detect-wileyfox-swift.txt and libmtp-1.1.14-new/._mtp-detect-wileyfox-swift.txt differ
Binary files libmtp-1.1.14/._mtp-detect-ziilabs-egg.txt and libmtp-1.1.14-new/._mtp-detect-ziilabs-egg.txt differ
Binary files libmtp-1.1.14/._mtp-detect-zune.txt and libmtp-1.1.14-new/._mtp-detect-zune.txt differ
Binary files libmtp-1.1.14/src/._libopenusb1-glue.c and libmtp-1.1.14-new/src/._libopenusb1-glue.c differ
diff -uNr libmtp-1.1.14/src/libopenusb1-glue.c libmtp-1.1.14-new/src/libopenusb1-glue.c
--- libmtp-1.1.14/src/libopenusb1-glue.c	2017-10-02 03:48:58.000000000 +0800
+++ libmtp-1.1.14-new/src/libopenusb1-glue.c	2017-11-17 14:44:07.295371082 +0800
@@ -1213,7 +1213,7 @@
     unsigned long packet_size;
     PTP_USB *ptp_usb = (PTP_USB *) params->data;
 
-    packet_size = ptp_usb->inep_maxpacket;
+    packet_size = ptp_usb->outep_maxpacket;
 
     LIBMTP_USB_DEBUG("SEND DATA PHASE\n");
 
diff -uNr libmtp-1.1.14/src/libusb1-glue.c libmtp-1.1.14-new/src/libusb1-glue.c
--- libmtp-1.1.14/src/libusb1-glue.c	2017-11-07 18:50:18.000000000 +0800
+++ libmtp-1.1.14-new/src/libusb1-glue.c	2017-11-17 14:44:02.441998744 +0800
@@ -52,15 +52,15 @@
  * spend a bit of time collecting data.  Higher values also
  * make connecting/disconnecting more reliable.
  */
-#define USB_START_TIMEOUT 15000
-#define USB_TIMEOUT_DEFAULT     20000
-#define USB_TIMEOUT_LONG        60000
+#define USB_START_TIMEOUT 50000
+#define USB_TIMEOUT_DEFAULT     50000
+#define USB_TIMEOUT_LONG        100000
 static inline int get_timeout(PTP_USB* ptp_usb)
 {
   if (FLAG_LONG_TIMEOUT(ptp_usb)) {
     return USB_TIMEOUT_LONG;
   }
-  return USB_TIMEOUT_DEFAULT;
+  return USB_TIMEOUT_LONG;
 }
 
 /* USB Feature selector HALT */
@@ -1010,7 +1010,7 @@
   PTPContainer MyEvent;
   unsigned long xread = 0;
   int old_callback_active = ptp_usb->callback_active;
-  int oldtimeout = 60000;
+  int oldtimeout = 600000;
 
 
   get_usb_device_timeout(ptp_usb, &oldtimeout);
@@ -1963,15 +1963,11 @@
     }
   }
 
-  /*
-   * It seems like on kernel 2.6.31 if we already have it open on another
-   * pthread in our app, we'll get an error if we try to claim it again,
-   * but that error is harmless because our process already claimed the interface
-   */
   usbresult = libusb_claim_interface(device_handle, ptp_usb->interface);
-
-  if (usbresult != 0)
-    fprintf(stderr, "ignoring libusb_claim_interface() = %d", usbresult);
+  if (usbresult != 0) {
+    fprintf(stderr, "error returned by libusb_claim_interface() = %d", usbresult);
+    return -1;
+  }
 
   /*
    * If the altsetting is set to something different than we want, switch
Binary files libmtp-1.1.14/src/._libusb-glue.c and libmtp-1.1.14-new/src/._libusb-glue.c differ
diff -uNr libmtp-1.1.14/src/libusb-glue.c libmtp-1.1.14-new/src/libusb-glue.c
--- libmtp-1.1.14/src/libusb-glue.c	2017-10-02 03:49:13.000000000 +0800
+++ libmtp-1.1.14-new/src/libusb-glue.c	2017-11-17 14:44:04.635364571 +0800
@@ -1205,7 +1205,7 @@
 	unsigned long packet_size;
 	PTP_USB *ptp_usb = (PTP_USB *) params->data;
 
-	packet_size = ptp_usb->inep_maxpacket;
+	packet_size = ptp_usb->outep_maxpacket;
 
 
 	LIBMTP_USB_DEBUG("SEND DATA PHASE\n");
Binary files libmtp-1.1.14/src/._music-players.h and libmtp-1.1.14-new/src/._music-players.h differ
diff -uNr libmtp-1.1.14/src/music-players.h libmtp-1.1.14-new/src/music-players.h
--- libmtp-1.1.14/src/music-players.h	2017-11-07 16:27:36.000000000 +0800
+++ libmtp-1.1.14-new/src/music-players.h	2017-11-17 14:44:00.241646774 +0800
@@ -628,6 +628,9 @@
   /* Reported by Jocelyn Mayer <l_indien@magic.fr> */
   { "Acer", 0x0502, "Iconia One 10", 0x3841,
       DEVICE_FLAGS_ANDROID_BUGS },
+  /* https://sourceforge.net/p/libmtp/bugs/1733/ */
+  { "Acer", 0x0502, "Liquid Zest 4G", 0x38a5,
+      DEVICE_FLAGS_ANDROID_BUGS },
   /* Mia */
   { "Acer", 0x0502, "Liquid Zest Plus", 0x38bb,
       DEVICE_FLAGS_ANDROID_BUGS },
@@ -1904,6 +1907,9 @@
   /* https://sourceforge.net/p/libmtp/support-requests/251/ */
   { "SONY", 0x0fce, "XPeria XZ Premium", 0x01f1,
       DEVICE_FLAG_NONE },
+  /* https://sourceforge.net/p/libmtp/support-requests/252/ */
+  { "SONY", 0x0fce, "XPeria XZ1 Compact", 0x01f4,
+      DEVICE_FLAG_NONE },
 
 
   /*
@@ -2023,6 +2029,8 @@
       DEVICE_FLAG_NONE },
   { "SONY", 0x0fce, "XPeria XZ Premium MTP+CDROM", 0x41f1,
       DEVICE_FLAG_NONE },
+  { "SONY", 0x0fce, "XPeria XZ1 Compact MTP+CDROM", 0x41f4,
+      DEVICE_FLAG_NONE },
 
   /*
    * MTP+ADB personalities of MTP devices (see above)
@@ -2165,6 +2173,8 @@
       DEVICE_FLAG_NONE },
   { "SONY", 0x0fce, "XPeria XZ Premium MTP+ADB", 0x51f1,
       DEVICE_FLAG_NONE },
+  { "SONY", 0x0fce, "XPeria XZ1 Compact MTP+ADB", 0x51f4,
+      DEVICE_FLAG_NONE },
 
   /*
    * MTP+UMS modes
@@ -2855,12 +2865,18 @@
   /* https://sourceforge.net/p/libmtp/bugs/1624/ */
   { "Lenovo", 0x17ef, "B Smartphone", 0x7a18,
       DEVICE_FLAGS_ANDROID_BUGS },
+  /* https://sourceforge.net/p/libmtp/bugs/1738/ */
+  { "Lenovo", 0x17ef, "K6 Power", 0x7a2a,
+      DEVICE_FLAGS_ANDROID_BUGS },
   /* https://sourceforge.net/p/libmtp/bugs/1679/ */
   { "Lenovo", 0x17ef, "C2", 0x7a36,
       DEVICE_FLAGS_ANDROID_BUGS },
   /* https://sourceforge.net/p/libmtp/support-requests/248/ */
   { "Lenovo", 0x17ef, "P2c72", 0x7a36,
       DEVICE_FLAGS_ANDROID_BUGS },
+  /* https://sourceforge.net/p/libmtp/bugs/1736/ */
+  { "Lenovo", 0x17ef, "P1060X", 0x9039,
+      DEVICE_FLAGS_ANDROID_BUGS },
 
   /*
    * Huawei
@@ -3124,6 +3140,8 @@
    * NEC
    */
   { "NEC", 0x0409, "FOMA N01A", 0x0242, DEVICE_FLAG_NONE },
+  /* https://sourceforge.net/p/libmtp/bugs/1724/ */
+  { "Casio", 0x0409, "GzOne Commando C771", 0x02ed, DEVICE_FLAG_NONE },
   { "NEC", 0x0409, "Casio C811", 0x0326, DEVICE_FLAG_NONE },
   { "NEC", 0x0409, "Casio CA-201L", 0x0432, DEVICE_FLAG_NONE },
 
@@ -3214,7 +3232,7 @@
   { "Amazon", 0x1949, "Kindle Fire 5", 0x0222,
       DEVICE_FLAGS_ANDROID_BUGS },
   /* https://sourceforge.net/p/libmtp/bugs/1718/ */
-  { "Amazon", 0x1949, "Kindle Fire 8", 0x0261,
+  { "Amazon", 0x1949, "Kindle Fire 8 (2nd ID)", 0x0261,
       DEVICE_FLAGS_ANDROID_BUGS },
   { "Amazon", 0x1949, "Fire Phone", 0x0800,
       DEVICE_FLAGS_ANDROID_BUGS },
@@ -3634,6 +3652,7 @@
   { "GoPro" , 0x2672, "HERO+", 0x0021, DEVICE_FLAG_NONE },
   { "GoPro" , 0x2672, "HERO5 Black", 0x0027, DEVICE_FLAG_NONE },
   { "GoPro" , 0x2672, "HERO5 Session", 0x0029, DEVICE_FLAG_NONE },
+  { "GoPro" , 0x2672, "HERO6 Black", 0x0037, DEVICE_FLAG_NONE },
 
   /* https://sourceforge.net/p/libmtp/bugs/1490/ */
   { "Marshall" , 0x2ad9, "London", 0x000b, DEVICE_FLAG_NONE },
